### PR TITLE
Display create RS and Gallery tool only for factious RS with same type

### LINKED ISSUE #4898: Gallery icon remains in fictitious record sets made using "Browse in Forms"
**Describe the bug**
The gallery icon was removed from fictitious record sets in the PR #4220. However, it was only removed from fictitious record sets created using the 'Add' button on an existing form, and not from those created using 'Browse in Forms' from a Query.

**To Reproduce**

1. Use a new or existing Query to query for records
2. Select more than one record
3. Click 'Browse in Forms' to create a fictitious record set
4. See gallery icon in the top right

**Expected behavior**
The gallery icon should be removed in order to be consistent with fictitious record sets made using the 'Add' button:
Screenshot 2024-05-08 at 4 53 05 PM

**Screenshots**

https://github.com/specify/specify7/assets/164079735/23b19d06-b928-4129-bfa7-1a2c1aa424d3

[Specify 7 System Information - 2024-05-08T22_03_10.639Z.txt](https://github.com/specify/specify7/files/15255457/Specify.7.System.Information.-.2024-05-08T22_03_10.639Z.txt)

Please, also fill out the following information manually:

- OS: Mac
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: Fitz_NHMD_Oct_2023
- Collection name: Vertebrate Paleontology
- User Name: petrifiedroadkill

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4898

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

- Go to data entry (ex CO)
- save form
- [ ] verify that Create RS tool are not available 
-  create a new data entry form using the same table as in first step (ex CO)
- [ ] verify that Gallery icon and Create RS tool are not available 
- save form 
- [ ] verify that Gallery icon and Create RS tool are available 
- create a new data entry form using another table than in first step (ex CE)
- [ ] verify that Gallery icon and Create RS tool are not available 
- save 
- [ ] verify that Gallery icon and Create RS tool are not available

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/FormSliders/RecordSet.tsx [hunk 1]
```diff
@@ -290,13 +290,25 @@ function RecordSet<SCHEMA extends AnySchema>({
...
   ): Promise<void> {
     if (!recordSet.isNew())
       await addIdsToRecordSet(resources.map(({ id }) => id));
     go(totalCount, resources[0].id, undefined, wasNew);
+    if (resourceType === undefined) {
+      setResourceType(resources[0].specifyTable.name);
+    } else if (resourceType !== resources.at(-1)?.specifyTable.name) {
+      setHasSeveralResourceType(true);
+      setResourceType(resources.at(-1)?.specifyTable.name);
+    }
```
Comment (sharadsw) [inline specifyweb/frontend/js_src/lib/components/FormSliders/RecordSet.tsx:311]: Do you need `resourceType` for anything apart from setting `hasSeveralResourceType` to true? If not, what do you think about just checking for `resources.length` instead?

Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/FormSliders/RecordSet.tsx:311]: An idea would have been to do: 
if (new Set(resources.map(resource => resource.name))).size > 1) setHasSeveralResourceType(true) 
But resources is actually an array of one. (the newly added resource) 
We do not have access to the previous resource table name unfortunately without saving it somehow

**Review by sharadsw — APPROVED** [review]

**Comment by maxpatiiuk** [comment]
> Display create RC and Gallery tool only for factious RC with same type

what does `RC` stand for?

Record Cet?

**Review by Areyes42 — CHANGES_REQUESTED** [review]
### Testing instructions

-   Go to data entry (ex CO)
-   save form
-   [ ] verify that Gallery icon and Create RS tool are not available
-   create a new data entry form using the same table as in first step (ex CO)
-   [x] verify that Gallery icon and Create RS tool are not available
-   save form
-   [x] verify that Gallery icon and Create RS tool are available
-   create a new data entry form using another table than in first step (ex CE)
-   [x] verify that Gallery icon and Create RS tool are not available
-   save
-   [x] verify that Gallery icon and Create RS tool are not available
***
Everything looks good, except saving the first CO still shows the Gallery icon as available. Is this intended behavior or a mistake in the testing instruction checklist?

https://github.com/specify/specify7/assets/130881183/415c5f2d-616e-4733-8a98-1949ae57d564

**Comment by CarolineDenis** [comment]
@Areyes42 , a mistake sorry for that

**Review by Areyes42 — APPROVED** [review]

**Review by lexiclevenger — APPROVED** [review]
### Testing instructions

- Go to data entry (ex CO)
- save form
- [x] verify that Create RS tool are not available 
-  create a new data entry form using the same table as in first step (ex CO)
- [x] verify that Gallery icon and Create RS tool are not available 
- save form 
- [x] verify that Gallery icon and Create RS tool are available 
- create a new data entry form using another table than in first step (ex CE)
- [x] verify that Gallery icon and Create RS tool are not available 
- save 
- [x] verify that Gallery icon and Create RS tool are not available 

---

Looks good! Everything is behaving according to the instructions.

**Review by combs-a — APPROVED** [review]
- Go to data entry (ex CO)
- save form
- [X] verify that Create RS tool are not available 
-  create a new data entry form using the same table as in first step (ex CO)
- [X] verify that Gallery icon and Create RS tool are not available 
- save form 
- [X] verify that Gallery icon and Create RS tool are available 
- create a new data entry form using another table than in first step (ex CE)
- [X] verify that Gallery icon and Create RS tool are not available 
- save 
- [X] verify that Gallery icon and Create RS tool are not available 

Looks good, seems to have fixed everything.

**Pull request merged**